pkg/device/device-util.go

```diff
@@ -551,3 +637,44 @@ func getPartitionPath(diskName string, partNum uint32) string {
 	}
 	return fmt.Sprintf("/dev/%s%d", diskName, partNum)
 }
+
+// getPartitionName returns the partition name from volume name
+func getPartitionName(volumeName string) string {
+	return strings.TrimPrefix(volumeName, "pvc-")
+}
+
+// parsePartedPartitionRow parses a single partition row in the output of parted command
+func parsePartedPartitionRow(partitionRow string) (partedOutput, error) {
+
+	// trim the ; from machine parseable output
+	partitionRow = strings.TrimSuffix(partitionRow, ";")
+
+	tmp := strings.Split(partitionRow, ":")
+
+	var partition partedOutput
+
+	partitionNumber, err := strconv.ParseInt(tmp[partedPartInfoPartNumIndex], 10, 32)
+	if err != nil {
+		return partition, fmt.Errorf("invalid partition number. row: %s, err: %v", partitionRow, err)
+	}
+
+	beginBytes, _ := strconv.ParseUint(tmp[partedPartInfoBeginBytesIndex][:len(tmp[partedPartInfoBeginBytesIndex])-1], 10, 64)
+	endBytes, _ := strconv.ParseUint(tmp[partedPartInfoEndBytesIndex][:len(tmp[partedPartInfoEndBytesIndex])-1], 10, 64)
+	size, _ := strconv.ParseUint(tmp[partedPartInfoSizeIndex][:len(tmp[partedPartInfoSizeIndex])-1], 10, 64)
```

Comment:
We should do error handling here rather than assuming the output to be in expected format. Please fill in the error wherever is missing in this file. We see one issue in our playground clusters where parted mkpart was invoked with empty disk string.

Reply:
Added the error check @iyashu